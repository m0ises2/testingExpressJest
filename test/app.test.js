const request = require('supertest');
const app = require('../app');

describe('Probar nuestro mini servidor', () => {
    test('Debe responder 200', (done) => {
        request(app).get('/').then(response => {
            expect(response.statusCode).toBe(200);
            done();
        });
    });

    test('Debe responder 404', async () => {
        const response = await request(app).get('/return404');
        expect(response.statusCode).toBe(404);
    });

    test('Retorna ok y objeto', async () => {
        const response = await request(app).post('/return/object');
        expect(response.statusCode).toBe(200);
        expect(response.body).toMatchSnapshot();

    })
});
