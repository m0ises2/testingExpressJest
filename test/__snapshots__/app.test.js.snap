// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar nuestro mini servidor Retorna ok y objeto 1`] = `
Object {
  "lastname": "Alvarado",
  "name": "Mois√©s",
}
`;
